manifests/profile/mistral.pp

```diff
@@ -246,26 +300,140 @@
+      true    => 'puppet:///modules/st2/etc/systemd/system/mistral.uwsgi.service',
+      default => 'puppet:///modules/st2/etc/systemd/system/mistral.service',
     }
-    'RedHat': {
-      file { '/etc/systemd/system/mistral.service':
-        ensure => file,
-        owner  => 'root',
-        group  => 'root',
-        mode   => '0444',
-        source => 'puppet:///modules/st2/etc/systemd/system/mistral.service',
+
+    case $::osfamily {
+      'Debian': {
+        # A bit sloppy, but this only covers Ubuntu right now. Fix this
+        file { '/etc/init/mistral.conf':
+          ensure => file,
+          owner  => 'root',
+          group  => 'root',
+          mode   => '0444',
+          source => $_upstart_init_script,
+        }
+      }
+      'RedHat': {
+        file { '/etc/systemd/system/mistral.service':
+          ensure => file,
+          owner  => 'root',
+          group  => 'root',
+          mode   => '0444',
+          source => $_systemd_init_script,
+        }
+      }
+      default: {
+        fail('[st2::profile::mistral] Unsupported Operatingsystem')
       }
     }
   }
   ### END Mistral Init Scripts ###
+
+  ### START uWSGI Specific Items ###
+  if $uwsgi {
+    $_sock_user = $::osfamily ? {
+      'Debian' => 'www-data',
+      'RedHat' => 'nginx',
+    }
+    $_socket = '/var/run/mistral_api.sock'
+    $_wsgi_file = '/opt/openstack/mistral/mistral/api/wsgi.py'
+
+    # WSGI file exists in 0.9 branch, so this is only temporary.
+    # JDF - 20150424
+    if $::st2::mistral_git_branch =~ /0.8/ {
+      file { $_wsgi_file:
+        ensure  => present,
+        owner   => 'root',
+        group   => 'root',
+        mode    => '0644',
+        source  => 'puppet:///modules/st2/mistral/wsgi.py',
+        require => Vcsrepo['/opt/openstack/mistral'],
+      }
+    }
+
+    ### Mistral API Settings
+    ini_setting { 'mistral_api_host':
+      ensure  => present,
+      path    => '/etc/mistral/mistral.conf',
+      section => 'api',
+      setting => 'host',
+      value   => $::st2::mistral_api_url,
+    }
+
+    ini_setting { 'mistral_api_port':
+      ensure  => present,
+      path    => '/etc/mistral/mistral.conf',
+      section => 'api',
+      setting => 'port',
+      value   => $::st2::mistral_api_port,
+    }
+
+    file { $_socket:
+      ensure => present,
+      owner  => $_sock_user,
+      group  => $_sock_user,
+    }
+
+    # Install a process supervisor. Using supervisord to assert
+    # a consistent management experience across systems, regardless
+    # if they are init.d, upstart, or systemd
+    include ::supervisord
+    ::python::pip { 'uwsgi':
+      ensure     => present,
+      virtualenv => $_system_python,
+    }
+
+    ::supervisord::program { 'mistral-uwsgi':
+      command  => "uwsgi -s ${_socket} --wsgi-file ${_wsgi_file} --chown-socket ${_sock_user}:${_sock_user} -H /opt/openstack/mistral/.venv/ -p ${uwsgi_processes} -l ${uwsgi_listen_queue_size}",
```

Comment:
What is the purpose of using supervisord for the uwsgi process and having separate configs for upstart/systemd?  Is this for Docker compatibility?  It seems it would lead to confusion by end users as to the proper way to manage the mistral api process.